runtime/recipe/slot-connection.js

```diff
@@ -85,7 +87,10 @@ class SlotConnection {
   }
 
   isResolved() {
-    // TODO: implement
+    if (!this.name || !this.particle || !this.targetSlot)
+      return false;
+    if (this.slotSpec.isRequired && this.targetSlot.sourceConnection == undefined)
```

Comment:
Dunno :)
== undefined is actually more precise, I think I just get lazy in places.